Working with forms (#2735)
umbraco/UmbracoDocs: Umbraco-Uno/Working-With-Forms/Setting-up-a-form/index.md

---
versionFrom: 8.0.0
---

# Setting up a form

> In this guide, you will learn how to set up a form in the backoffice. When you get an Uno project, it comes with the unicorn hotel template, Which includes two forms. However, suppose those forms dont meet your needs, then you can create your own.

Reviewer: ```suggestion
In this guide, you will learn how to set up a form in the Umbraco Uno backoffice. 

When you get an Uno project, it comes with the Unicorn Hotel and Spa template, which includes two forms. However, suppose those forms don't meet your needs, then you can create your own.
```

> * First, you will have to navigate to the forms section in the back office.
> * Then when in the forms section, in the tree on the left, there's a folder icon next to that it says ***Forms*** hover this and click the ***...*** button that shows up.
> * A list will show up. On that list, select Empty form.

Reviewer: ```suggestion
* Navigate to the Forms section in the backoffice.
* In the tree on the left, hover ***Forms*** and select the ***...***
* On the list that shows up, select "Empty form"
```

![This image shows the forms section and where to press for a new form](images/Forms-backoffice.png)

> ## Adding the content of the form

Reviewer: ```suggestion
## Building the form
```

Now we have taken the first step to creating our very own form. In the next steps, we will name the form and add the fields that it needs to contain.

![An image showing the main view of the new form](images/Back-form.png)

> * Enter the forms name where it says ***Enter Name***.

Reviewer: ```suggestion
* Give the new form a name where it says ***Enter Name***.
```

> * You will see that the form already contains an item. This is the permission to store data, so if you need this for the form you are creating, you can go ahead and leave that. Otherwise, you can delete it by clicking the bin icon at the top of the field.

Reviewer: I think this should be a note instead 🤔 What do you think?

---
versionFrom: 8.0.0
---

# Setting up a form

In this guide, you will learn how to set up a form in the Umbraco Uno backoffice. 

> When you get your Uno project, it will not have any forms, so you will have to make them your self which can be a challenge when you have never done it before. In this guide, you will learn how to build a form from scratch.

Reviewer: ```suggestion
When you get your Uno project, it will not have any forms, so you will have to make them yourself. In this guide, you will learn how to build a form from scratch.
```

> If this is your first form, you will have a green button in the middle of the screen that says ***Create form*** if this is not your first form, proceed to the next steps.

Reviewer: Shouldn't this sentence be **before** the first step in the following list?

```suggestion
If this is your first form, you will have a green button in the middle of the screen that says ***Create form***. If this is not your first form, proceed to the next steps.
```

* Navigate to the Forms section in the backoffice.
* In the tree on the left, hover ***Forms*** and select the ***...***
* On the list that shows up, select "Empty form."

![This image shows the forms section and where to press for a new form](images/Forms-backoffice.png)

## Building the form

Now we have taken the first step to creating our very own form. In the next steps, we will name the form and add the fields that it needs to contain.

![An image showing the main view of the new form](images/Back-form.png)

:::note
You will see that the form already contains an item. This is the permission to store data, so if you need this for the form you are creating, you can go ahead and leave that. Otherwise, you can delete it by clicking the bin icon at the top of the field.
:::

* Give the new form a name where it says ***Enter Name***.

> * Now to adding a group Click where it says ***Add new group***.

Reviewer: You will only have this option if you chose to remove the "Consent" field, right?

```suggestion
* Now to add a group click where it says ***Add new group***.
```

Author: i believe you have it no matter what

Reviewer: Ah, OK! So, the guide shows how to create fields that are in another Group than the Consent field.
I missed that.

* Follow up by clicking ***Add Question***.

> * Name the first question, "Name" and chose a ***short answer*** for the answer type.

Reviewer: ```suggestion
* Name the first question, "Name" and choose the ***Short Answer*** as the answer type.
```

> * In the settings further down, put it to mandatory and add the text to show if this mandatory field is not filled in.

Reviewer: ```suggestion
* In the settings further down, set it to mandatory and add a text to show if this field is not filled in.
```

* When that is done, hit the green submit button at the bottom.

![This image shows Settings of the fields in the form](images/Settings-form.png)

## Finalizing the form

> Now that we have done this supposes, you have more than one group of items. However, you might want to rearrange them. This can be done by hitting the ***Reorder*** button and dragging the groups in the order you want.

Reviewer: I'm not entirely sure I understand the first sentence here.
What exactly are you trying to say? 🤔

Author: In this guide, you will learn how to set up a form in the Umbraco Uno backoffice.  this?

It is just to let them know that they will learn how to set up a form 🤔

Author: ohhh :s here i just want to let them know how to use the reorder option

:::tip
It can be a good idea to name the groups to keep track of what each group does. This can be done by clicking where it says ***Untitled group***.
:::

> Now you can go ahead and add what you like to the form, and when you are done, hit save. Once that is done, you will have a new custom made form to add to your widgets.
> In the next artice you can learn to [add the Umbraco Forms widget](../Umbraco-Forms-widget).

Reviewer: ```suggestion
Now you can go ahead and add what you like to the form, and when you are done, hit save. Once that is done, you will have a new custom made form to add to your widgets.

In the next article you can learn how to add forms to your website by using [he Umbraco Forms widget](../Umbraco-Forms-widget).
```
